### DISCUSSION TITLE
Need a way for corrosion to ignore `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER` flags

### DISCUSSION THREAD

**Discussion by Billyzou0741326** [corrosion-rs/corrosion#533] [Q&A] [answered]
My use case requires that part of my project be compiled by `qt`'s llvm, taking the following flags: 
```
"-DCMAKE_C_COMPILER:FILEPATH=C:/Qt/Tools/llvm-mingw1706_64/bin/clang.exe"
"-DCMAKE_CXX_COMPILER:FILEPATH=C:/Qt/Tools/llvm-mingw1706_64/bin/clang++.exe"
```

but this messes up compilation of the other part of the project that's built by rust, configured by corrosion (note that it's trying to use the `qt`'s llvm instead of whichever one rustc normally uses),

```
...
   Compiling static_assertions v1.1.0
   Compiling version_check v0.9.4
error: linking with `C:/Qt/Tools/llvm-mingw1706_64/bin/clang.exe` failed: exit code: 1
  |
  = note: "C:/Qt/Tools/llvm-mingw1706_64/bin/clang.exe" ...
  = note: clang-17: warning: argument unused during compilation: '-no-pie' [-Wunused-command-line-argument]
          lld: error: unable to find library -lgcc_eh
          lld: error: unable to find library -lgcc
          clang-17: error: linker command failed with exit code 1 (use -v to see invocation)
```

which ends up failing to find the necessary libraries.

Is there any way to sidestep this problem?  (It feels like I'd either have to fail the rust side compilation or the qt compilation)

---

*Edit*: What I'm trying to achieve is, to keep using the `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER` flags for Qt compilation, but at the same time have corrosion takes care of compiling the rust module- without using the compiler specified for Qt. Would that be possible?

**Comment by jschwe** [comment, marked as answer]
What corrosion does is instructing `rustc` to use the same **linker** as CMake would choose. The default linker (or linker-driver) in rust depends on the target, but usually it is just whatever is `cc`.  You can explicitly set a different linker by calling `corrosion_set_linker` (See also https://corrosion-rs.github.io/corrosion/usage.html#per-target-options).

Probably you don't even need to choose a different linker though, I suspect you are just missing some linker arguments like `--sysroot=${CMAKE_SYSROOT}`

**Reply by Billyzou0741326** [reply]
Thanks for the pointers.  Apparently the problem lies in Qt's `llvm-mingw` compiler doesn't come with `libgcc` and friends - seems like it uses `libclang_rt` instead.

I was able to compile with Qt's `mingw` gcc / g++ compiler just fine (since it has the correct libs), with the linker corrosion selects by default.
